File: apps/auth-service/app/src/main/java/ml/ipvz/fa/authservice/config/SecurityConfig.java
package ml.ipvz.fa.authservice.config;

import ml.ipvz.fa.authservice.model.config.TokenConfig;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity;
import org.springframework.security.config.web.server.ServerHttpSecurity;
import org.springframework.security.web.server.SecurityWebFilterChain;

@Configuration
@EnableWebFluxSecurity
public class SecurityConfig {

    @Bean
    @ConfigurationProperties(prefix = "token")
    public TokenConfig item() {
        return new TokenConfig();
    }


    @Bean
    public SecurityWebFilterChain filterChain(ServerHttpSecurity http) {
        return http
                .csrf().disable()
                .httpBasic().disable()
                .build();
    }


}

File: apps/group-service/app/src/main/kotlin/ml/ipvz/fa/groupservice/config/SecurityConfig.kt
package ml.ipvz.fa.groupservice.config

import ml.ipvz.fa.authservice.base.config.BaseSecurityConfig
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
import org.springframework.security.config.web.server.ServerHttpSecurity
import org.springframework.security.web.server.SecurityWebFilterChain
import org.springframework.security.web.server.context.ServerSecurityContextRepository

@Configuration
@Import(BaseSecurityConfig::class)
class SecurityConfig(
    private val serverSecurityContextRepository: ServerSecurityContextRepository
) {

    @Bean
    fun filterChain(http: ServerHttpSecurity): SecurityWebFilterChain {
        return http
            .httpBasic().disable()
            .formLogin().disable()
            .csrf().disable()
            .logout().disable()

            .securityContextRepository(serverSecurityContextRepository)

            .authorizeExchange()
            .pathMatchers("/actuator/health").permitAll() //health
            .anyExchange().authenticated()

            .and()
            .build()
    }
}